### ISSUE TITLE
Can't install fedn on Python 3.10

### ISSUE THREAD

**Issue by benblamey** [scaleoutsystems/fedn#393]
**Severity**
- [ ] Critical/Blocker (select if the issue makes the application unusable or causes serious data loss)
- [ ] High (select if the issue affects a major feature and there is no workaround or the available workaround is very complex)
- [x] Medium (select if the issue affects a minor feature or affects a major feature but has an easy enough workaround to not cause any major inconvenience)
- [ ] Low (select if the issue doesn't significantly affect the user experience, like minor visual bugs)

**Describe the bug**
My first issue is that fedn/tree/master/examples/mnist-keras/bin/init_venv.sh calls `python`. This is broken on my machine, but I can use `python3`. (Will `python` always point to python3.x ?)

Anyway, here is the issue:

python3 -m pip install -e ../../fedn
Gives:
ERROR: Package 'fedn' requires a different Python: 3.10.4 not in '<3.9,>=3.8'

This is in the setup.py:
python_requires='>=3.8,<3.9',

If I fix that, it then needs to compile grpc from source, because I'm on a mac, possibly due to this issue: https://github.com/grpc/grpc/issues/23509

This is very slow, and it ultimately fails anyway, due to this issue:
https://github.com/grpc/grpc/issues/24677
 
**Environment:**
 - OS: MacOS 12.3.1 (on an intel i5 processor).
 - Version: [e.g. macOS Catalina]

**Reproduction Steps**

**Expected behavior**

**Screenshots**

**Contact Details**
An e-mail address in case we need to contact you for further details regarding this issue.

**Comment by benblamey** [comment]
So two bugs here, second one is that because of the grpc issues, I can't run fedn on my mac...

**Comment by evertrol** [comment]
Is this still an issue?
The grpcio has long been solved, since version 1.39 at least (following the second link above; that issue has been closed there). Changing the version to match 1.39.0 (and Python to 3.10) installs fine for me, on macOS 12.4. (I haven't tried running it! So I don't know where the 1.34 requirement comes from, and what 1.39 or later will result in.)
Perhaps time to change the grpcio version requirement (and Python version; 3.11 is coming up as well), so that this issue can be closed?

**Comment by ahellander** [comment]
Yes, I agree. The older grpcio dep was because some users really wanted to be able to use older versions of tensorflow (and still run fedn in the same env). But I think we should bump the dependencies now, and if someone wants to use old tensorflows they need to manage that by isolating the environments for client and server (which would be the case in a realistic deployment anyways).

**Comment by ahellander** [comment]
This one has been resolved now in develop and will be in the 0.4 release.
